webpack/components/extensions/HostDetails/Tabs/PackagesTab.js

```diff
@@ -0,0 +1,123 @@
+import './PackagesTab.scss';
+
+export const PackagesTab = () => {
+  const hostDetails = useSelector(state => selectAPIResponse(state, 'HOST_DETAILS'));
+  const { id: hostId } = hostDetails;
+  const actionButtons = <Button isDisabled> {__('Upgrade')} </Button>;
+
+  const [searchQuery, updateSearchQuery] = useState('');
+
+  const emptyContentTitle = __('This host does not have any packages.');
+  const emptyContentBody = __('Packages will appear here when available.');
+  const emptySearchTitle = __('No matching packages found');
+  const emptySearchBody = __('Try changing your search settings.');
+  const columnHeaders = [
+    __('Package'),
+    __('Status'),
+    __('Installed Version'),
+    __('Upgradable To'),
+  ];
+
+  const fetchItems = useCallback(
+    params => getInstalledPackagesWithLatest(hostId, params),
+    [hostId],
+  );
+
+  const response = useSelector(state => selectAPIResponse(state, HOST_PACKAGES_KEY));
+  const { results, ...metadata } = response;
+  const status = useSelector(state => selectHostPackagesStatus(state));
+  const rowActions = [
+    {
+      title: __('Upgrade via remote execution'), disabled: true,
+    },
+    {
+      title: __('Upgrade via customized remote execution'), disabled: true,
+    },
+  ];
+
+  return (
+    <div>
+      <div id="packages-hint">
+        <Hint>
+          <HintBody>
+            {__('Packages management functionality on this page is incomplete')}.
+            <br />
+            <Button component="a" variant="link" isInline href={urlBuilder(`content_hosts/${hostId}/packages/installed`, '')}>
+              {__('Visit the previous Packages page')}.
+            </Button>
+          </HintBody>
+        </Hint>
+      </div>
+      <div id="packages-tab">
+        <TableWrapper
+          {...{
+                metadata,
+                emptyContentTitle,
+                emptyContentBody,
+                emptySearchTitle,
+                emptySearchBody,
+                status,
+                actionButtons,
+                searchQuery,
+                updateSearchQuery,
+                }
+          }
+          additionalListeners={[hostId]}
+          fetchItems={fetchItems}
+          autocompleteEndpoint={`/hosts/${hostId}/packages/auto_complete_search`}
+          foremanApiAutoComplete
+          variant={TableVariant.compact}
+        >
+          <Thead>
+            <Tr>
+              {columnHeaders.map(col =>
+                <Th key={col}>{col}</Th>)}
+              <Th />
+            </Tr>
+          </Thead>
+          <Tbody>
+            {results?.map((packages) => {
+                const {
+                  id,
+                  name: packageName,
+                  nvra: installedVersion,
+                } = packages;
+                return (
+                  <Tr key={`${id}`}>
+                    <Td>
+                      <a href={urlBuilder(`packages/${id}`, '')}>{packageName}</a>
```

Comment:
From what I can tell these ids are the HostInstalledPackage ID rather than a Katello::Rpm. Remember that a HostInstalledPackage can possibly (and probably not uncommonly) have no corresponding Katello::Rpm. That is that case for me in my dev env cause I don't have any repos synced that my client host also has installed packages for. 

AFAIK the fix would be to 1) not have a link when there is no corresponding Katello::Rpm (Content -> Packages) for the given HostInstalledPackage and 2) Use the Katello::Rpm ID here in the URL rather than the HostInstalledPackage ID